### PR TITLE
Identifiers resolution fixes (1.3)

### PR DESCRIPTION
**Pull request by IvanSavenko**
- VCMI will now correctly resolve identifiers in unexpected form. For example, if hero spellbook has entry "spell.bless" vcmi will now correctly load this identifier as "bless" of type "spell" instead of searching for spell named "spell.bless"
- Fixed moat identifiers requests - will now always request identifiers of type "spell". Updated config to use short form.
- Fixed invalid empty scope on resolving map overrides

### CODE REVIEW

File: lib/spells/effects/Moat.cpp [hunk 1]
```diff
@@ -129,7 +129,6 @@ void Moat::placeObstacles(ServerCallback * server, const Mechanics * m, const Ef
 {
 	assert(m->battle()->battleGetDefendedTown());
 	assert(m->casterSide == BattleSide::DEFENDER); // Moats are always cast by defender
-	assert(turnsRemaining < 0); // Moats should lasts infininte number of turns
```
[the remark this answers is not in the document]
Comment (IvanSavenko) [reply lib/spells/effects/Moat.cpp:132]: Moat.cpp:132:11: error: 'turnsRemaining' is a private member of 'spells::effects::Obstacle'
Obstacle.h:67:10: note: declared private here

File: lib/CTownHandler.cpp [hunk 2]
```diff
@@ -885,10 +885,13 @@ void CTownHandler::loadTown(CTown * town, const JsonNode & source)
 		town->namesCount += 1;
 	}
 
-	VLC->modh->identifiers.requestIdentifier(source["moatAbility"], [=](si32 ability)
+	if (!source["moatAbility"].isNull()) // VCMI 1.2 compatibility code
```
Comment (rilian-la-te) [inline lib/CTownHandler.cpp:888]: If there is no moatAbility, moats will not work.

File: lib/spells/effects/Moat.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (IvanSavenko) [reply lib/spells/effects/Moat.cpp:132]: gcc
Plus integrated into IDE clang.

Are you using debug configuration?

[the remark this answers is not in the document]
Comment (rilian-la-te) [reply lib/spells/effects/Moat.cpp:132]: RelWithDebInfo almost always.

[the remark this answers is not in the document]
Comment (IvanSavenko) [reply lib/spells/effects/Moat.cpp:132]: And I always use Debug, except for when I need profilling runs. So asserts trigger for me.

**Pull request merged**